packages/block-library/src/file/index.js

```diff
@@ -77,20 +77,27 @@ export const settings = {
 		from: [
 			{
 				type: 'files',
-				isMatch: ( files ) => files.length === 1,
+				isMatch( files ) {
+					return files.length > 0;
+				},
 				// We define a lower priorty (higher number) than the default of 10. This
 				// ensures that the File block is only created as a fallback.
 				priority: 15,
 				transform: ( files ) => {
-					const file = files[ 0 ];
-					const blobURL = createBlobURL( file );
+					const blocks = [];
 
-					// File will be uploaded in componentDidMount()
-					return createBlock( 'core/file', {
-						href: blobURL,
-						fileName: file.name,
-						textLinkHref: blobURL,
+					files.map( ( file ) => {
+						const blobURL = createBlobURL( file );
```

Comment:
Out of scope of this PR but I thought I'd mention for context. Ideally, we shouldn't rely on the `edit` function to properly upload the file and use the final URL. Instead we should avoid storing temporary attributes in blocks. 

That said, we don't have a proper solution for this problem at the moment, we'll more likely need to think about a "reactive" transform API but this is not high-priority.